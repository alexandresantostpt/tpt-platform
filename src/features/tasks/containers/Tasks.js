import { connect } from 'react-redux'

import * as actionsTask from '../actions'

import { getListSelector } from '../selectors'

import Tasks from '../pages/Tasks'

const mapStateToProps = state => ({
    tasks: getListSelector(state)
})

const mapDispatchToProps = {
    ...actionsTask
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Tasks)
