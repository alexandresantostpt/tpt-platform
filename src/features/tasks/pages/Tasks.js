import PropTypes from 'prop-types'
import React, { useEffect, useState } from 'react'
import styled from 'styled-components'

import { i18n } from '@i18n'

import CollapseDivider from '@components/collapse/CollapseDivider'
import If from '@components/if/If'
import Title from '@components/title/Title'

import { getUser } from '@utils/auth'
import { isEmpty, not } from '@utils/functions'

import AddButton from '../components/AddButton'
import Task from '../components/task/Task'

const StyledTasks = styled.div`
    overflow-y: auto;
`

const Tasks = props => {
    const { tasks } = props
    const [newTask, updateNewTask] = useState(false)
    const user = getUser()

    useEffect(() => {
        const { requestGetAllAction } = props
        requestGetAllAction()
    }, [])

    const handleAddTask = () => updateNewTask(true)

    const del = id => {
        const { requestDelAction } = props
        requestDelAction({ id: id })
    }

    const stopEditing = (describe, task) => {
        const { requestSaveDataAction } = props
        not(isEmpty(describe)) && requestSaveDataAction({ ...task, describe: describe, user: task.user || user.id })
        newTask && updateNewTask(false)
    }

    const switchTaskStatus = task => {
        const { requestSaveDataAction } = props
        requestSaveDataAction({ ...task, done: not(task.done) })
    }

    const tasksToDo = tasks.filter(item => not(item.done))
    const tasksDone = tasks.filter(item => item.done)

    return (
        <StyledTasks>
            <Title>{i18n.t('titles.tasks')}</Title>
            <AddButton handleAddTask={handleAddTask} />
            <If condition={newTask}>
                <Task describe="" newTask stopEditing={stopEditing} />
            </If>
            {tasksToDo.map(task => (
                <Task
                    describe={task.describe}
                    key={task._id}
                    removeTask={del}
                    stopEditing={stopEditing}
                    switchTaskStatus={switchTaskStatus}
                    task={task}
                />
            ))}
            <If condition={!!tasksDone.length}>
                <CollapseDivider initialState={false} title={`ConcluÃ­das (${tasksDone.length})`}>
                    {tasksDone.map(task => (
                        <Task
                            describe={task.describe}
                            key={task._id}
                            removeTask={del}
                            stopEditing={stopEditing}
                            switchTaskStatus={switchTaskStatus}
                            task={task}
                        />
                    ))}
                </CollapseDivider>
            </If>
        </StyledTasks>
    )
}

Tasks.defaultProps = {
    tasks: []
}

Tasks.propTypes = {
    requestDelAction: PropTypes.func.isRequired,
    requestSaveDataAction: PropTypes.func.isRequired,
    tasks: PropTypes.array
}

export default Tasks
