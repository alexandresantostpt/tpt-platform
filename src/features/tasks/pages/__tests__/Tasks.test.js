import configureMockStore from 'redux-mock-store'
import React from 'react'
import renderer from 'react-test-renderer'

import { List, Map, Record } from 'immutable'
import { Provider } from 'react-redux'

import Tasks from '../Tasks'

describe('Tests for Travel component', () => {
    let mockStore = {}
    let store = {}
    let toast = {}

    const tasks = [
        {
            __v: 28,
            _id: '5c9e31f8c57b4917b426e334',
            createdAt: '2019-03-29T14:55:52.764Z',
            describe: 'teste 123',
            done: false,
            updatedAt: '2019-03-29T14:55:52.764Z',
            user: '5c9dfb65d1cf42bd5fa66b3d'
        },
        {
            __v: 2,
            _id: '5c9e31f8c57b4917b426e344',
            createdAt: '2019-03-29T14:55:52.764Z',
            describe: 'teste 1234',
            done: true,
            updatedAt: '2019-03-29T14:55:52.764Z',
            user: '5c9dfb65d1cf42bd5fa66b3d'
        }
    ]

    beforeEach(() => {
        toast = { icon: 'check', message: 'success', type: 'success' }

        mockStore = configureMockStore()
        store = mockStore(
            Record({
                toastReducers: Map({
                    list: List.of(toast)
                })
            })()
        )
    })

    it('Should render withou crash', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Tasks
                        requestDelAction={jest.fn()}
                        requestGetAllAction={jest.fn()}
                        requestSaveDataAction={jest.fn()}
                        tasks={tasks}
                    />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
