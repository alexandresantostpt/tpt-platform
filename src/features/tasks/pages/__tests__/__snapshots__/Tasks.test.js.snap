// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Travel component Should render withou crash 1`] = `
<div
  className="Tasks__StyledTasks-sc-1hy8ads-0 cjVtaH"
>
  <h2
    className="Title__StyledTitle-sc-11bxhsf-0 frSblw"
  >
    Tarefas
  </h2>
  <button
    className="AddButton__StyledAddButon-ppf0uf-0 bLjgkJ"
    onClick={[Function]}
  >
    <span
      className="icon icon-plus Icon__StyledIcon-crr1cg-0 eXMubC"
      disabled={false}
      fontSize="0.8rem"
      onClick={null}
    />
    Adicionar Tarefa
  </button>
  <div
    className="Task__StyledTask-sc-1fi2258-0 cEDnPQ"
  >
    <span
      className="TaskCheck__StyledTaskCheck-sc-1tgftv-0 htWEWq"
      onClick={[Function]}
    >
      <button
        className="TaskCheck__StyledButton-sc-1tgftv-1 elQAyk"
      />
      <span
        className="icon icon-check Icon__StyledIcon-crr1cg-0 cmjVJQ"
        disabled={false}
        fontSize="1.75rem"
        onClick={null}
      />
    </span>
    <div
      className="TaskBody__StyledBody-sc-1p8y2k3-0 cOetdt"
    >
      <textarea
        autoComplete="off"
        className="TaskEdit__StyledTaskEdit-f5yune-0 dWmTjV"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        value="teste 123"
      />
    </div>
    <span
      className="TaskActions__StyledTaskActions-sc-92wkgs-0 oqREy"
    >
      <span
        className="icon icon-times Icon__StyledIcon-crr1cg-0 dqcjNO"
        disabled={false}
        fontSize="1.5rem"
        onClick={[Function]}
      />
    </span>
  </div>
  <div
    className="CollapseDivider__StyledCollapseDivider-sc-1otxrtt-0 dEjhoP"
  >
    <div
      className="CollapseDivider__StyledHeader-sc-1otxrtt-1 hTeopB"
    >
      Conclu√≠das (1)
      <span
        className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 cMlgiC"
        disabled={false}
        fontSize="1.75rem"
        onClick={[Function]}
      />
    </div>
    <div
      className="CollapseItems__StyledCollapseItems-tyowld-0 kEpAYd"
    >
      <div
        className="Task__StyledTask-sc-1fi2258-0 cEDnPQ"
      >
        <span
          className="TaskCheck__StyledTaskCheck-sc-1tgftv-0 bcxwQS"
          onClick={[Function]}
        >
          <button
            className="TaskCheck__StyledButton-sc-1tgftv-1 elQAyk"
          />
          <span
            className="icon icon-check Icon__StyledIcon-crr1cg-0 cmjVJQ"
            disabled={false}
            fontSize="1.75rem"
            onClick={null}
          />
        </span>
        <div
          className="TaskBody__StyledBody-sc-1p8y2k3-0 cOetdt"
        >
          <textarea
            autoComplete="off"
            className="TaskEdit__StyledTaskEdit-f5yune-0 ihYtDu"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            value="teste 1234"
          />
        </div>
        <span
          className="TaskActions__StyledTaskActions-sc-92wkgs-0 oqREy"
        >
          <span
            className="icon icon-times Icon__StyledIcon-crr1cg-0 dqcjNO"
            disabled={false}
            fontSize="1.5rem"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
